### ISSUE TITLE
Fix a menuitem labeling omission in the menu pattern

### ISSUE THREAD

**Issue by mcking65** [w3c/aria-practices#215]
If a menuitem element contains or owns a child menu and its menuitems, then the menuitem needs either aria-label or aria-labelledby similar to what is described for treeitems. This is not covered in the roles, states, and properties section of the [menu design pattern](http://w3c.github.io/aria-practices/#menu).

The [editor menubar example](http://w3c.github.io/aria-practices/examples/menubar/menubar-2/menubar-2.html) demonstrates this situation.

**Comment by mcking65** [comment]
Proposed fix for this issue can be seen in a branch in rawgit at:

Changes include:
* Added statement: "If activating a menuitem opens a submenu, the menuitem is known as a parent menuitem. A submenu's menu element is contained in or owned by it's parent menuitem."
* Added statement: "A parent menuitem has either aria-label or aria-labelledby set to specify a label. Warning: parent menuitems missing a label are labeled by all child elements, causing screen readers to announce every child menuitem and often making a menu so verbose that it is unusable."
* Simplified statement to use parent menuitem terminology: "A parent menuitem has aria-haspopup set to true."

**Comment by mcking65** [comment]
@annabbott, @jnurthen, 

Please review the proposed fix for this issue, which can be seen in a branch in rawgit at:
https://rawgit.com/w3c/aria-practices/menu-issues-215-261/aria-practices.html#menu

Changes include:

* Added statement: "If activating a menuitem opens a submenu, the menuitem is known as a parent menuitem. A submenu's menu element is contained in or owned by it's parent menuitem."
* Added statement: "A parent menuitem has either aria-label or aria-labelledby set to specify a label. Warning: parent menuitems missing a label are labeled by all child elements, causing screen readers to announce every child menuitem and often making a menu so verbose that it is unusable."
* Simplified statement to use parent menuitem terminology: "A parent menuitem has aria-haspopup set to true."

**Comment by annabbott** [comment]
See my comment in https://github.com/w3c/aria-practices/issues/261. 

Under Keyboard interaction > Down arrow > as last bullet:
When focus is in a menubar, does nothing.

Under Keyboard interaction > Escape
Escape: Close the menu that contains focus and return focus to the element or context, e.g., menu button or parent menuitem, from which the menu was opened. 
Suggest removing reference to 'menu button' and replace with 'menu'.

**Comment by mcking65** [comment]
Ann, for this fix I didn't change the keyboard subsection. I changed only the above mentioned bullets in the states and properties.

Nonetheless, to address your specific concern ...

@annabbott wrote:
>Under Keyboard interaction > Down arrow > as last bullet:
>When focus is in a menubar, does nothing.
>Under Keyboard interaction > Escape
>Escape: Close the menu that contains focus and return focus to the element or context, e.g., menu button or parent menuitem, from which the menu was opened. 
>Suggest removing reference to 'menu button' and replace with 'menu'.

We don't want to say to return focus to a menu. A menu does not receive focus. If it ever receives DOM focus, then it needs to be using aria-activedescendant to place AX focus on a menuitem. 

There are two ways of opening an ARIA menu:

1. Open it by activating a menuitem. 
2. open it by activating a menubutton.

Note that a menuitem must be in either a menubar or a parent menu.

**Comment by annabbott** [comment]
ok

**Comment by jnurthen** [comment]
Modify the 2nd change to the following:
If a parent menu has descendent items in the DOM then set a label on the parent using aria-label or aria-labelledby. Parent menuitems with descendent menuitems missing a label are labeled by all child elements, causing screen readers to announce every child menuitem and often making a menu so verbose that it is unusable. If using aria-owns this is not necessary.

**Comment by jnurthen** [comment]
I have always thought this was the way it was but we should probably do a
little test case to check. I won't be able to do it this weekend though.
aria-owns is not mentioned at all in the accessible name calculation. All
the definitions of nodes in the accessible name calculation are DOM nodes,
so my assumption is that child nodes means child DOM nodes - but it could
do with being clarified.

**Comment by mcking65** [comment]
This issue depends on issue #277.

**Comment by mcking65** [comment]
I am going to merge these changes and close this issue. If the outcome of issue #277 calls for more changes, thenI will make them at that time.

**Comment by mcking65** [comment]
Changes merged. Note that I made an error in the title of commit cbe304b. It says it is for issues 2155 and 266. It should say it is for issues 215 and 261. Sorry about that.
